packages/phoenix-event-display/src/loaders/objects/phoenix-objects.ts

```diff
@@ -479,38 +490,45 @@ export class PhoenixObjects {
   /**
    * Get the position for a Calo hit in cartesian coordinates
    * @param clusterParams Parameters for the Cluster (which must include theta and phi)
-   * @param drawRadius Radius where to draw barrel Clusters
-   * @param maxR2 Maximum permitted radius squared
-   * @param maxZ Maximum position along the z axis
+   * @param defaultRadius Default cylindrical radius (rho) where to draw barrel Clusters
+   * @param defaultZ Default position along the z axis
    * @returns Corrected cartesian position.
    */
   private static getCaloPosition(
-    clusterParams: { phi: number; eta: number; theta: number },
-    drawZ: number,
-    drawRadius: number,
-    maxR2: number,
-    maxZ: number
+    clusterParams: {
+      phi: number;
+      eta: number;
+      radius?: number;
+      z?: number;
+      theta?: number;
+    },
+    defaultRadius: number = 1800,
+    defaultZ: number = 3600
   ) {
-    const theta = CoordinateHelper.etaToTheta(clusterParams.eta);
-    clusterParams.theta = theta;
+    const theta =
+      clusterParams.theta ?? CoordinateHelper.etaToTheta(clusterParams.eta);
+    const radius = clusterParams.radius ?? defaultRadius;
 
     const position = CoordinateHelper.sphericalToCartesian(
-      drawZ + drawRadius,
+      radius,
       theta,
       clusterParams.phi
     );
 
-    // How to generalise to other experiments? Pass in limit lambda?
-    const cylRadius2 = position.x * position.x + position.y * position.y;
```

Comment:
I think this is the reason for #552 @kjvbrt ... you removed the cylindrical cut didn't you? Or am I missing something?